.palantir/revapi.yml

```diff
@@ -1,4 +1,85 @@
+      new: "class org.apache.iceberg.SerializableTable"
+      justification: "Serialization changed"
+    - code: "java.class.defaultSerializationChanged"
+      old: "class org.apache.iceberg.TableMetadata"
+      new: "class org.apache.iceberg.TableMetadata"
+      justification: "Serialization changed"
+    - code: "java.field.noLongerConstant"
+      old: "field org.apache.iceberg.TableProperties.DELETE_MODE_DEFAULT"
+      new: "field org.apache.iceberg.TableProperties.DELETE_MODE_DEFAULT"
+      justification: "Refactorings"
+    - code: "java.field.noLongerConstant"
+      old: "field org.apache.iceberg.TableProperties.MERGE_MODE_DEFAULT"
+      new: "field org.apache.iceberg.TableProperties.MERGE_MODE_DEFAULT"
+      justification: "Refactorings"
+    - code: "java.field.noLongerConstant"
+      old: "field org.apache.iceberg.TableProperties.UPDATE_MODE_DEFAULT"
+      new: "field org.apache.iceberg.TableProperties.UPDATE_MODE_DEFAULT"
+      justification: "Refactorings"
+    - code: "java.method.addedToInterface"
+      new: "method <T extends org.apache.iceberg.rest.RESTResponse> T org.apache.iceberg.rest.RESTClient::delete(java.lang.String,\
+        \ java.util.Map<java.lang.String, java.lang.String>, java.lang.Class<T>, java.util.Map<java.lang.String,\
+        \ java.lang.String>, java.util.function.Consumer<org.apache.iceberg.rest.responses.ErrorResponse>)"
+      justification: "Adding new method where query params can be used"
+    - code: "java.method.addedToInterface"
+      new: "method org.apache.hadoop.conf.Configuration org.apache.hadoop.conf.Configurable::getConf()\
+        \ @ org.apache.iceberg.hadoop.HadoopConfigurable"
+      justification: "New method was added to org.apache.hadoop.conf.Configurable"
+    - code: "java.method.addedToInterface"
+      new: "method void org.apache.hadoop.conf.Configurable::setConf(org.apache.hadoop.conf.Configuration)\
+        \ @ org.apache.iceberg.hadoop.HadoopConfigurable"
+      justification: "New method was added to org.apache.hadoop.conf.Configurable"
+    - code: "java.method.numberOfParametersChanged"
+      old: "method void org.apache.iceberg.ManifestReader<F extends org.apache.iceberg.ContentFile<F\
+        \ extends org.apache.iceberg.ContentFile<F>>>::<init>(org.apache.iceberg.io.InputFile,\
+        \ java.util.Map<java.lang.Integer, org.apache.iceberg.PartitionSpec>, org.apache.iceberg.InheritableMetadata,\
+        \ org.apache.iceberg.ManifestReader.FileType)"
+      new: "method void org.apache.iceberg.ManifestReader<F extends org.apache.iceberg.ContentFile<F\
+        \ extends org.apache.iceberg.ContentFile<F>>>::<init>(org.apache.iceberg.io.InputFile,\
+        \ int, java.util.Map<java.lang.Integer, org.apache.iceberg.PartitionSpec>,\
+        \ org.apache.iceberg.InheritableMetadata, org.apache.iceberg.ManifestReader.FileType)"
+      justification: "Avoids reading ManifestFile when creating ManifestReader"
```

Comment:
For this one, can we restore and deprecate the old constructor? This was an optimization to avoid opening a file because we could pass in the spec ID. We can support both though.

Reply:
I've restored the original code for the old constructor + deprecated it